src/async_wrap.cc

```diff
@@ -176,6 +176,10 @@ void AsyncWrap::EmitAfter(Environment* env, double async_id) {
 
 class PromiseWrap : public AsyncWrap {
  public:
+  enum InternalFields {
+    kIsChainedPromiseField = AsyncWrap::InternalFields::kCount,
```

Comment:
You can skip the `::InternalFields` everywhere, right?

(I’d also rename `kCount` → `kInternalFieldCount`)

Reply:
> You can skip the ::InternalFields everywhere, right?

Yeah, and I go back and forth on it. I tend to prefer being extra precise here but have no problem removing it.

> (I’d also rename kCount → kInternalFieldCount)

👍

Reply:
I think the fact that you can then drop the extra enums for subclasses is a good reason to not include it :)

If you *do* want an enum that’s always accessed through its explicit name, then `enum class` might be better anyway (but I don’t think that’s necessary here, the classes should provide enough namespacing)